### DISCUSSION TITLE
What tools do I need in order to build libduckdb-sys on Ubuntu Server?

### DISCUSSION THREAD

**Discussion by derekmahar** [dathere/qsv#804] [FAQ]
What tools do I need in order to build dependency `libduckdb-sys` on Ubuntu Server?  I have installed packages build-essential, clang, and cmake.  What am I missing?

**Comment by jqnatividad** [comment]
@derekmahar , I gather you've been able to build `qsv` on Ubuntu Server with the libduckdb-sys dependency.

Can you share what you needed to install to do so?

**Comment by derekmahar** [comment]
From a brand new installation of Ubuntu Server 22.04, to build `qsv`, I had to install [rustup](https://rustup.rs/) and Ubuntu packages `build-essential`, `clang`, `cmake`, and `python3.10-dev` which Cargo required in order to build dependency `libduckdb-sys` in particular.

```
sudo apt install build-essential clang cmake python3.10-dev
```

**Reply by derekmahar** [reply]
I'm surprised that I didn't have to install a Lua development package. Does the `qsv` build not require any Lua C or C++ header files?

**Reply by jqnatividad** [reply]
Nope.  The awesome [`mlua`](https://github.com/khvzak/mlua) crate handles all the Lua/Luau dependencies. It uses https://github.com/khvzak/luau-src-rs to build Luau and embed it.
